fastapi/app/main.py

```diff
@@ -0,0 +1,239 @@
+"""Module providing SpiceQL endpoints"""
+
+from ast import literal_eval
+from typing import Annotated, Any
+from fastapi import FastAPI, Query
+from pydantic import BaseModel, Field
+from starlette.responses import RedirectResponse
+import numpy as np
+import pyspiceql
+
+SEARCH_KERNELS_BOOL = True
+
+# Models
+class MessageItem(BaseModel):
+    message: str
+
+class ResultModel(BaseModel):
+    result: Any = Field(serialization_alias='return')
+
+class ErrorModel(BaseModel):
+    error: str
+
+class ResponseModel(BaseModel):
+    statusCode: int
+    body: ResultModel | ErrorModel
+
+# Create FastAPI instance
+app = FastAPI()
+
+# General endpoints
+@app.get("/", include_in_schema=False)
+async def root():
+    return RedirectResponse(url="/docs")
+
+@app.post("/customMessage")
+async def message(
+    message_item: MessageItem
+    ):
+    return {"message": message_item.message}
+
+
+# SpiceQL endpoints
+@app.get("/getTargetStates")
+async def getTargetStates(
+    target: str,
+    observer: str,
+    frame: str,
+    abcorr: str,
+    mission: str,
+    ets: Annotated[list[float], Query()] | str | None = None,
+    startEts: float | None = None,
+    exposureDuration: float | None = None,
+    numOfExposures: int | None = None,
```

Comment:
Maybe Im missing something, but isn't it better to make a new endpoint rather than adding to the existing one? Im not 100% sure what would be the better practice.

Reply:
I thought so too, but I've tried that and found that FastAPI doesn't seem to support having duplicate endpoints with varying query params. Seems like the general consensus was to have one endpoint with either optional params and data validation ([example](https://stackoverflow.com/a/74759329)) or create two models (one `ets` only, other with `startEts` etc.) where the input data can be validated that way ([example](https://stackoverflow.com/a/75460395)).

Reply:
I think this approach isn't bad 🤔 thinking about it over the weekend. I think I prefer it over another endpoint with a different name.